src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -7,9 +7,52 @@
 public class TypewiseAlertTest 
 {
     @Test
-    public void infersBreachAsPerLimits()
+    public void infersBreachToolow()
     {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+      assertTrue(TypewriteAlert.inferBreach(12, 20, 30) ==
+        BreachTypes.TOO_LOW);
     }
+
+    @Test
```

Comment:
there are totally 2 targets, 3 cooling types , and 5 values (2 boundary , 1 low, 1 normal, 1 very high), which means there could be potentially 30 variations. Please check if important variations are considered.